### PR TITLE
don't send instance and zone if values not specified

### LINKED ISSUE #55: Don't send instance or zone if there is no value.
Don't send instance or zone if there is no value.

### PR DESCRIPTION
**Pull request by nolanmar511**
Fixes #55

### CODE REVIEW

File: ts/src/config.ts [hunk 1]
```diff
@@ -87,8 +87,8 @@ export interface ProfilerConfig extends AuthenticationConfig {
   projectId?: string;
   logLevel: number;
   serviceContext: {service: string; version?: string;};
-  instance: string;
-  zone: string;
+  instance?: string;
```
Comment (aalexand) [inline ts/src/config.ts]: I would keep the type and skip sending if the string is empty - what's wrong with that? We don't want to send empty strings either and I am not sure the current code does not?

Comment (nolanmar511) [reply ts/src/config.ts]: Right now, the code will skip sending these if empty.
I've added a test for that now.

File: ts/src/index.ts [hunk 2]
```diff
@@ -84,11 +84,11 @@ export async function initConfig(config: Config): Promise<ProfilerConfig> {
                 (err: Error) => {
                     // ignore errors, which will occur when not on GCE.
                 }) ||
-        ['', ''];
-    if (!mergedConfig.zone) {
+        [undefined, undefined];
```
Comment (ofrobots) [inline ts/src/index.ts]: `[]` would be adequate.

Comment (nolanmar511) [reply ts/src/index.ts]: I tried that, but got the following error:

ts/src/index.ts(78,12): error TS2525: Initializer provides no value for this binding element and the binding element has no default value.

78     const [instance, zone] =

**Comment by nolanmar511** [comment]
PTAL

**Comment by aalexand** [comment]
@nolanmar511 Merge this?

**Comment by aalexand** [comment]
The description should say that this fixes https://github.com/GoogleCloudPlatform/cloud-profiler-nodejs/issues/55 ?

**Pull request merged**